### DISCUSSION TITLE
Commit with version tag not found

### DISCUSSION THREAD

**Discussion by Crossbow78** [adamralph/minver#865] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/7549753/230742583-93866f6b-f561-4fe8-8c0b-45262f6b400e.png)
In this picture the two topmost commits are in a feature branch, the rest is in master branch.

As you can see I tagged one (latest) commit in `master` branch, and when the pipeline builds that commit it produces nuget package `MyOwnLibrary-0.1.0.nupkg` as expected.

Now I'm in my feature branch, and run my pipeline on latest commit. My expectation is that it would arrive at version `0.2.0-alpha.0.2` where the last 2 is the height to the tagged commit. Btw, this would however cause a conflict between two feature branches that are at the same distance...
But it does not find the tagged commit at all as can be seen in below logs, what am I missing here?

![image](https://user-images.githubusercontent.com/7549753/230742603-24cae9b1-3b4a-4fbc-ae40-a6da9f3a806e.png)

**Comment by Crossbow78** [comment]
Here's a more detailed log... it seems that `git log` has a smaller reach than expected. Does anyone know why?

![image](https://user-images.githubusercontent.com/7549753/231156148-ee539616-a6d2-4c50-9c0c-bad9d32f23ee.png)

(Note that commit 8573d5c7fd130733a6b36b2655342b28fdcad284 is the new latest commit in that feature branch where I changed the MinVer log level)

**Comment by Crossbow78** [comment, marked as answer]
This seems to be caused by a [recent change in default Azure DevOps pipeline behavior](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines#shallow-fetch) where the git checkout has changed to a shallow fetch with maxDepth 1. The issue also affected [Nerdbank.Gitversioning](https://github.com/dotnet/Nerdbank.GitVersioning/blob/main/doc/cloudbuild.md#azure-pipelines).

When adding the following step in my pipeline, Minver is behaving exactly as expected:
```
- checkout: self
  fetchDepth: 0
```

**Comment by adamralph** [comment]
FTR this has since been added to the [FAQs](https://github.com/adamralph/minver?tab=readme-ov-file#why-is-the-default-version-sometimes-used-in-github-actions-azure-pipelines-and-travis-ci-when-a-version-tag-exists-in-the-history).
